src/new_hope/week1.clj

```diff
@@ -50,3 +134,100 @@
     (crit/quick-bench (second (reverse coll)))
     (crit/quick-bench (last (butlast coll))))
   )
+
+;; Sum it all up
+(def sumseq (fn [xs]
+              (reduce + xs)))
+
+(facts "about sumseq"
+  (sumseq [1 2 3]) => 6
+  (sumseq (list 0 -2 5 5)) => 8
+  (sumseq #{4 2 1}) => 7
+  (sumseq '(0 0 -1)) => -1
+  (sumseq '(1 10 3)) => 14)
+
+;; Find the odd numbers
+(def onlyodd (fn [xs]
+               (filter odd? xs)))
+
+(facts "about onlyodd"
+  (onlyodd #{1 2 3 4 5}) => '(1 3 5)
+  (onlyodd [4 2 1 6]) => '(1)
+  (onlyodd [2 2 4 6]) => '()
+  (onlyodd [1 1 1 3]) => '(1 1 1 3))
+
+;; Palindrome Detector
+(def palindrome? (fn [s]
+                   (= (seq s) (reverse s))))
+
+(fact "about palindrom?"
+  (palindrome? '(1 2 3 4 5)) => false
+  (palindrome? "racecar") => true
+  (palindrome? [:foo :bar :foo]) => true
+  (palindrome? '(1 1 3 3 1 1)) => true
+  (palindrome? '(:a :b :c)) => false)
+
+;; Duplicate a Sequence
+(def dupe (fn [xs]
+            (reduce concat (map #(repeat 2 %) xs))))
+
+(facts "about dupe"
+  (dupe [1 2 3]) => '(1 1 2 2 3 3)
+  (dupe [:a :a :b :b]) => '(:a :a :a :a :b :b :b :b)
+  (dupe [[1 2] [3 4]]) => '([1 2] [1 2] [3 4] [3 4]))
+
+;; Day 5
+;; Compress a Sequence
+(defn removedupes [xs]
+  (reduce (fn [xs x]
+            (if (not= (last xs) x) (conj xs x) xs))
```

Comment:
Keep in mind that `last` is based on the `seq` abstraction that has no notion of length, it is inefficient in this case.
Theoretically, this can be implemented using a complex accumulator that contains the previous value as well: [
In this case the easiest way is just to use collection-aware functions: `conj`, `peek`, `pop`. With vectors they all work on the right end.